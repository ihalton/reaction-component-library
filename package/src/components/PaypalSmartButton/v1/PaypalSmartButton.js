import React, { Component, Fragment } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { withComponents } from "@reactioncommerce/components-context";
import { PayPalButton } from "react-paypal-button-v2";

const PaypalButtonWrapper = styled.div`
  -webkit-font-smoothing: antialiased;
  box-sizing: border-box;
  display: flex;
  flex-direction: row;
  position: relative;
  min-width: 800px;
`;

class PaypalSmartButton extends Component {
  static propTypes = {
    /**
     * If you've set up a components context using
     * [@reactioncommerce/components-context](https://github.com/reactioncommerce/components-context)
     * (recommended), then this prop will come from there automatically. If you have not
     * set up a components context or you want to override one of the components in a
     * single spot, you can pass in the components prop directly.
     */
    components: PropTypes.shape({}).isRequired,

    amount: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    currency: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    shippingPreference: PropTypes.string,
    onSuccess: PropTypes.func,
    catchError: PropTypes.func,
    onError: PropTypes.func,
    createOrder: PropTypes.func,
    createSubscription: PropTypes.func,
    onApprove: PropTypes.func,
    style: PropTypes.object,
    options: PropTypes.shape({
      clientId: PropTypes.string,
      merchantId: PropTypes.string,
      currency: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
      intent: PropTypes.string,
      commit: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
      vault: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
      component: PropTypes.string,
      disableFunding: PropTypes.string,
      disableCard: PropTypes.string,
      integrationDate: PropTypes.string,
      locale: PropTypes.string,
      buyerCountry: PropTypes.string,
      debug: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
    }),
    onButtonReady: PropTypes.func,
  };

  static defaultProps = {
    style: {},
    options: {
      clientId: "sb",
      currency: "USD",
    },
    shippingPreference: "GET_FROM_FILE",
  };

  state = {};

  componentDidUpdate = () => {};

  render() {
    const {createOrder, onApprove, options} = this.props;
    return (
      <PaypalButtonWrapper>
        <PayPalButton options={options} createOrder={(data, actions) => createOrder(data, actions)} onApprove={(data, actions) => onApprove(data, actions)}></PayPalButton>
      </PaypalButtonWrapper>
    );
  }
}

export default withComponents(PaypalSmartButton);
