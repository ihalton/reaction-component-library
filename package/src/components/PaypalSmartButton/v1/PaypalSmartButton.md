### Overview



### Usage



### Theme

Assume that any theme prop that does not begin with "rui" is within `rui_components`. See [Theming Components](./#!/Theming%20Components).

| Theme Prop                      | Default                                                      | Description                                  |
| ------------------------------- | ------------------------------------------------------------ | -------------------------------------------- |
| `Input.backgroundColor_dark`    | white                                                        | Background color when `isDarkBackground`     |
| `Input.backgroundColor_default` | black02                                                      | Background color when not `isDarkBackground` |
| `Input.borderColor_default`     | black20                                                      | Border color in "default" state              |
| `Input.borderColor_error`       | red                                                          | Border color in "error" state                |
| `Input.borderColor_focus`       | teal                                                         | Border color in "focus" state                |
| `Input.borderColor_success`     | teal                                                         | Border color in "success" state              |
| `Input.color_default`           | coolGrey500                                                  | Input text color when in "default" state     |
| `Input.color_disabled`          | black25                                                      | Input text color when in "disabled" state    |
| `Input.color_error`             | red                                                          | Input text color when in "error" state       |
| `Input.color_focus`             | coolGrey500                                                  | Input text color when in "focus" state       |
| `Input.color_success`           | black55                                                      | Input text color when in "success" state     |
| `Input.fontFamily`              | `'Source Sans Pro', 'Helvetica Neue', Helvetica, sans-serif` | Input text font                              |
| `Input.fontSize`                | 14px                                                         | Input text font size                         |
| `Input.horizontalPadding`       | 10px                                                         | Left and right padding                       |
| `Input.lineHeight`              | 1                                                            | Input line height                            |
| `Input.placeholderColor`        | black20                                                      | Placeholder text color                       |
| `Input.verticalPadding`         | 8px                                                          | Top and bottom padding                       |

#### Typography

None
