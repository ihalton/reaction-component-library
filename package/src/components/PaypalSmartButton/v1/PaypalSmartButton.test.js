import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { ComponentsProvider } from "@reactioncommerce/components-context";
import mockComponents from "../../../tests/mockComponents";
import realComponents from "../../../tests/realComponents";
import PaypalSmartButton from "./PaypalSmartButton";

test("basic snapshot", () => {
    const component = renderer.create(<PaypalSmartButton components={mockComponents} />);
  
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });