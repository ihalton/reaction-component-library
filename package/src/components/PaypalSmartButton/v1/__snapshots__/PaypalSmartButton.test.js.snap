// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  position: relative;
  min-width: 800px;
}

<div
  className="c0"
/>
`;
